MODULE_ACCEPTANCE_CRITERIA.MD

```diff
@@ -90,6 +93,7 @@ It is also important to note that while this document provides the criteria agai
   * _This is not applicable to libraries_
 * Must not depend on a FOLIO library that has not been approved through the TCR process
 * Gracefully handles the absence of third party systems or related configuration. (3, 5, 12)
-* Sonarqube hasn't identified any security issues, major code smells or excessive (>3%) duplication (6)
+* Sonarqube hasn't identified any security issues, major code smells or excessive (>3%) duplication (6); and any disabled rules are reasonably justified.
+  * note: guidelines on how to disable individual rules, with justification. (@TODO create wiki page / link) 
```

Comment:
I think this should be revised a little to also account for situations where rules aren't disabled but the submitters have actively decided not to fix a sonar violation. Sometimes the rule makes sense and we don't want to disable it, but it doesn't necessarily apply to a specific instance.

One example is the max class hierarchy depth of 6. This seems like a good sonar rule, but it's overly strict in some cases, like exceptions, where every exception class starts 3-4 levels deep right off the bat (we should make an exception for exceptions 🙂)

Maybe something along the lines of this (but hopefully reworded, since this change feels a little clunky):
```suggestion
* Sonarqube hasn't identified any security issues, major code smells, or excessive (>3%) duplication (6); and any disabled rules or intentionally ignored recommendations are reasonably justified.
  * note: guidelines on how to disable individual rules, with justification. (@TODO create wiki page / link) 
```

Reply:
This should ideally apply to `@SuppressWarning` annotations, `// NOSONAR` comments, and sonar exclusions in the POM

Reply:
Thanks @julianladisch that works great for my second question (and folks can edit it further as needed).  I will wait to put that link in place until we have consensus on the other question as well:

> Where do you imagine the justification being provided. In the code itself? (If so, would there be some common string that the reviewers could search for?) Or in the TCR template?

Reply:
@mweaver-ebsco if you agree with @julianladisch that the justification should be in the code itself, then I think we do need a common search string for the TCR evaluator(s) to find them, so we can review if they are "reasonably justified".  Suggestions @mweaver-ebsco @julianladisch ?

Reply:
I agree with Julian, that we don't really need a common string. The ways to ignore a violation in code already provide a small number of things to search for ("NOSONAR," "SuppressWarnings," and "sonar" should catch pretty much everything), and the explanation should be near that (or in the cases where the violation was marked as resolved in Sonar as "won't fix," there's a comment field where the justification can be added).

Reply:
I'm going to adjust the language to more-or-less match @mweaver-ebsco 's suggestion above.

Assuming this PR is merged, I will do a separate PR on the dev.folio.org repo to be clear that these justifications are now required instead of "might be useful".

Reply:
I've drafted that PR, to be merged after this one.  @julianladisch please see the request there to add justification to those examples.